sdks/unity/Tests/Runtime/PlayMode/AgonesSdkIntegrationTests.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2024 Google LLC
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Agones;
+using System.Threading.Tasks;
+using System;
+
+namespace Tests.Runtime.Agones
+{
+    public class AgonesSdkIntegrationTests
+    {
+
+        private AgonesSdk sdk;
+        private AgonesBetaSdk betaSdk;
+        private AgonesAlphaSdk alphaSdk;
+
+        [UnitySetUp]
+        public IEnumerator UnitySetUp()
+        {
+            GameObject gameObject = new GameObject();
+            gameObject.AddComponent<AgonesSdk>();
+            gameObject.AddComponent<AgonesAlphaSdk>();
+            gameObject.AddComponent<AgonesBetaSdk>();
+            yield return null;
+
+            this.sdk = gameObject.GetComponent<AgonesSdk>();
+            this.betaSdk = gameObject.GetComponent<AgonesBetaSdk>();
+            this.alphaSdk = gameObject.GetComponent<AgonesAlphaSdk>();
+
+            Assert.IsNotNull(this.sdk);
+            Assert.IsNotNull(this.betaSdk);
+            Assert.IsNotNull(this.alphaSdk);
+        }
+
+        [UnityTest]
+        public IEnumerator TestSdk()
+        {
+            var task = RunSdkTests();
+            yield return new WaitUntil(() => task.IsCompleted);
+
+            if (task.Exception != null)
+            {
+                Debug.LogError(task.Exception);
+                Assert.Fail();
+            }
+        }
+
+        private async Task RunSdkTests()
+        {
+            var connected = await this.sdk.Connect();
+            Assert.IsTrue(connected);
+
+            bool hasReceivedUpdates = false;
+            this.sdk.WatchGameServer((gs) =>
+            {
+                hasReceivedUpdates = true;
+                Assert.NotNull(gs.ObjectMeta);
+                Assert.NotNull(gs.Status);
+                Assert.NotNull(gs.Spec);
+            });
+
+            // Run tests
+            var ready = await this.sdk.Ready();
+            Assert.IsTrue(ready);
+
+            var setLabel = await this.sdk.SetLabel("label", "test_label");
+            Assert.IsTrue(setLabel);
+
+            var setAnnotation = await this.sdk.SetAnnotation("annotation", "test_annotation");
+            Assert.IsTrue(setAnnotation);
+
+            var reserved = await this.sdk.Reserve(TimeSpan.FromSeconds(5));
+            Assert.IsTrue(reserved);
+
+            await Task.Delay(1000);
```

Comment:
Just curious why the delay?

Reply:
I assume to simulate reservation occurring sometime before the server is allocated.